cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,710 @@
+
+
+def _compute_subtype_lqs(
+    labels,
+    predictions,
+    *,
+    alpha,
+    low_probability_threshold,
+    high_probability_threshold,
+    temperature,
+):
+    """
+    Returns a label quality score for each datapoint.
+    Score is between 0 and 1.
+
+    1 - clean label (given label is likely correct).
+    0 - dirty label (given label is likely incorrect).
+
+    Parameters
+    ----------
+    labels:
+        A list of `N` dictionaries for `N` images such that `labels[i]` contains the given labels for the `i`-th image in the format
+       `{'bboxes': np.ndarray((M,4)), 'labels': np.ndarray((M,)), 'image_name': str}` where `M` is the number of annotated bounding boxes
+       for the `i`-th image and `bboxes[j]` is in the format [x,y,x,y] with given label `labels[j]`. ('image_name' is optional here)
+
+    predictions:
+        A list of `N` `np.ndarray` for `N` images such that `predictions[i]` corresponds to the model predictions for the `i`-th image
+        in the format `np.ndarray((K,))` where K is the number of classes and `predictions[i][k]` is of shape `np.ndarray(M,5)`
+        where `M` is the number of predicted bounding boxes for class `K` and the five columns correspond to `[x,y,x,y,pred_prob]` returned
+        by the model.
+
+        Note: `M` number of predicted bounding boxes can be different from `M` number of annotated bounding boxes for class `K` of `i`-th image.
+
+    alpha:
+        Weight between IoU and distance when considering similarity matrix. High alpha means considering IoU more strongly over distance.
+
+    low_probability_threshold:
+        The lowest prediction threshold allowed when considering predicted boxes to identify badly located label boxes.
+
+    high_probability_threshold:
+        The high probability threshold for considering predicted boxes to identify overlooked and swapped label boxes.
+
+    temperature:
+        Temperature of the softmin function where a lower score suggests softmin acts closer to min.
+
+    Returns
+    ---------
+    label_quality_scores:
+        Array of shape ``(N, )`` of scores between 0 and 1, one per image in the dataset.
+        Lower scores indicate images are more likely to contain an incorrect label.
+    """
+    scores = []
+
+    min_possible_similarity = _get_min_possible_similarity(predictions, labels, alpha)
+    for prediction, label in zip(predictions, labels):
+        lab_bboxes, lab_labels = _separate_label(label)
+        pred_bboxes, pred_labels, pred_label_prob = _separate_prediction(prediction)
+        pred_label_prob = np.array(pred_label_prob)
+        iou_matrix = _get_overlap_matrix(lab_bboxes, pred_bboxes)
+        dist_matrix = 1 - _get_dist_matrix(lab_bboxes, pred_bboxes)
+
+        similarity_matrix = iou_matrix * alpha + (1 - alpha) * (1 - dist_matrix)
+        assert (similarity_matrix.flatten() >= 0).all() and (similarity_matrix.flatten() <= 1).all()
+
+        scores_overlooked = []
```

Comment:
can you have separate methods to separately compute:
`scores_overlooked` , `scores_swap` , `scores_badloc`  `scores_swap`

Or is there some object reuse that makes this challenging?
Making them separate methods is much cleaner code and easier to expose to users in the future.

Would be nice to also have separate helper methods that take in these scores and aggregate them into one S_swap, S_badloc, S_overlooked score per image. Again that will be nice for users to have access to